components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/utils/mappings/throttling/CommonThrottleMappingUtil.java

```diff
@@ -205,33 +205,65 @@ public static ThrottleConditionDTO fromConditionToDTO(Condition condition)   //.
      * @return Derived Quota policy object from DTO
      * @throws UnsupportedThrottleLimitTypeException
      */
-    public static QuotaPolicy fromDTOToQuotaPolicy(ThrottleLimitDTO dto) throws UnsupportedThrottleLimitTypeException {
+    public static QuotaPolicy fromDTOToQuotaPolicy(ThrottleLimitTypeDTO dto)
+            throws UnsupportedThrottleLimitTypeException {
+
+        ThrottleLimitDTO throttleLimit = getThrottleLimitType(dto);
         QuotaPolicy quotaPolicy = new QuotaPolicy();
-        quotaPolicy.setLimit(fromDTOToLimit(dto));
-        quotaPolicy.setType(mapQuotaPolicyTypeFromDTOToModel(dto.getType()));
+        quotaPolicy.setLimit(fromDTOToLimit(throttleLimit));
+        quotaPolicy.setType(mapQuotaPolicyTypeFromDTOToModel(throttleLimit.getType()));
         return quotaPolicy;
     }
 
     /**
-     * Converts a Quota Policy object into a Throttle Limit DTO object
+     * Obtain Throttle Limit DTO object from Throttle Limit Type object
+     *
+     * @param dto Throttle Policy Default limit DTO object
+     * @return Throttle Limit DTO object
+     * @throws UnsupportedThrottleLimitTypeException
+     */
+    public static ThrottleLimitDTO getThrottleLimitType(ThrottleLimitTypeDTO dto)
+            throws UnsupportedThrottleLimitTypeException {
+
+        if (dto.getBandwidth() != null && dto.getRequestCount() != null) {
+            String msg = "Throttle limit types " + dto.getBandwidth().getClass().getName() + " and " +
+                    dto.getRequestCount().getClass().getName() + " cannot be specified at once";
+            log.error(msg);
+            throw new UnsupportedThrottleLimitTypeException(msg);
+        } else if (dto.getBandwidth() != null) {
+            return dto.getBandwidth();
+        } else if (dto.getRequestCount() != null) {
+            return dto.getRequestCount();
+        } else {
+            String msg = "A Throttle limit type has not been specified";
+            log.error(msg);
```

Comment:
same here

Reply:
throw new UnsupportedThrottleLimitTypeException("A Throttle limit type has not been specified");